packages/namekit-react/src/components/NamePrice.tsx

```diff
@@ -0,0 +1,82 @@
+import { DomainName, Price, Registration } from "@namehash/ens-utils";
+
+import React from "react";
+import { DisplayedPrice } from "./DisplayedPrice";
+import { RegistrationPriceInfo } from "./RegistrationPriceInfo";
+import {
+  AltPriceDisplayFormat,
+  PriceDisplayPosition,
+  PriceDisplaySize,
+  PriceSymbolPosition,
+  PriceSymbology,
+} from "./PriceSymbol";
+import { RegistrationPriceInfoDisplayingFormat } from "./RegistrationPriceInfo";
+
+interface NamePriceProps {
+  registration: Registration;
+  parsedName: DomainName;
+  // The price to be displayed
+  price: Price;
+  // The alternative price to be displayed (the price converted to another currency)
+  altPrice?: Price;
+  // The onClick event handler for the price text
+  onTextClick?: () => void;
+  // Wether to display the alternative price as a text instead of a tooltip
+  altPriceDisplayFormat: AltPriceDisplayFormat;
+  // Wether to display a currency symbology as its acronym or symbol
+  currencySymbology?: PriceSymbology;
+  // The place to display the currency symbology
+  symbolPosition?: PriceSymbolPosition;
+  // The size of the alternative price display
+  altPriceDisplaySize?: PriceDisplaySize;
+  // The size of the price display
+  priceTextDisplaySize?: PriceDisplaySize;
+  // The place to display the alternative price when displayed as text
+  altPriceDisplayPosition?: PriceDisplayPosition;
+  // Wether or not to display the name of the currency in a tooltip when its symbol is hovered
+  showCurrencyTooltipDescription?: boolean;
+}
+
+export const NamePrice = ({
```

Comment:
I'm not sure if this UI component should even exist? If we do keep it I'm open to that, but then it should be made more generic.

For example, when I test this UI component in the storybook it is always right aligned, there's a big rectangle with that highlights with a grey background on hover, there's a clicky-hand cursor, etc.. These ideas shouldn't be in this UI component if it is generic.

We need to put more special care and attention into how we build our own abstractions. We should try to avoid creating abstractions that mix ideas from too many different layers at the same time. For example here we're defining a UI component that has too much of an expectation that it is only placed inside a DomainCard.